lib/system/database.rb

```diff
@@ -7,36 +7,31 @@ module Database
     class ConnectionError < ActiveRecord::NoDatabaseError; end
     module_function
 
-    def configuration_specification
-      @configuration_specification ||= read_configuration_specification
-    end
-
-    def read_configuration_specification
-      configurations = Rails.application.config.database_configuration
-      resolver = ActiveRecord::ConnectionAdapters::ConnectionSpecification::Resolver.new(configurations)
-      spec = ActiveRecord::ConnectionHandling::DEFAULT_ENV.call
-
-      resolver.spec(resolver.configurations[spec])
+    def database_config
+      @database_config ||= begin
+        configurations = Rails.application.config.database_configuration
+        ActiveRecord::DatabaseConfigurations.new(configurations).configs_for(env_name: Rails.env).first
+      end
     end
 
     def adapter
       adapter_method.match(/^(oracle|postgres|mysql).*/).to_a.last
     end
 
     def adapter_method
-      ActiveSupport::StringInquirer.new(configuration_specification.adapter_method)
+      @adapter_method ||= ActiveSupport::StringInquirer.new(database_config.adapter)
```

Comment:
`database_config.adapter_method` also exists, but it basically is a contatenation of `_connection` to the adapter value:
https://github.com/rails/rails/blob/v6.1.7.7/activerecord/lib/active_record/database_configurations/database_config.rb#L27-L29